Source/WebKit/WPEPlatform/wpe/WPEBufferDMABuf.cpp

```diff
@@ -150,45 +157,51 @@ static gpointer wpeBufferDMABufImportToEGLImage(WPEBuffer* buffer, GError** erro
 }
 
 #if USE(GBM)
-static struct gbm_device* gbmDevice()
+static bool wpeBufferDMABufTryEnsureGBMDevice(WPEBufferDMABuf* buffer)
 {
-    static struct gbm_device* device;
-    static std::once_flag onceFlag;
-    std::call_once(onceFlag, [] {
-        const char* filename = wpe_render_node_device();
-        if (!filename)
-            return;
-
-        int fd = open(filename, O_RDWR | O_CLOEXEC);
-        if (fd == -1)
-            return;
-
-        device = gbm_create_device(fd);
-        if (device)
-            return;
-
-        close(fd);
-    });
-    return device;
+    auto* priv = buffer->priv;
```

Comment:
Is it possible that buffer or priv are NULL? Maybe we can add an assertion to check it?

Reply:
No, this is is called from a public method that already checks buffer is a `WPE_BUFFER_DMA_BUF`